src/actions/receive-profile.js

```diff
@@ -185,8 +185,11 @@ export function finalizeProfileView(
         break;
       case 'active-tab':
         if (selectedThreadIndex === null) {
-          // Switch back over to the full view.
-          timelineTrackOrganization = { type: 'full' };
+          // Switch back over to the full view if selectedThreadIndex is not present.
+          // We check this here because if selectedThreadIndex is null, that means
+          // it's a new profile from Firefox directly and has no profile information
+          // encoded in the URL. But we only allow conversions from full view currently.
+          dispatch(finalizeFullProfileView(profile, selectedThreadIndex));
```

Comment:
@julienw I tried your suggestion about moving this `if` to above of this switch block. But that way, we still need to check `selectedThreadIndex === null` for active tab view here, because flow wants this to be non-null. So instead of having two same if checks in different places (or using ensureExists), I felt like dispatching full view here is better.

Reply:
Good for me! but then please pass `null` directly instead of `selectedThreadIndex` in this branch :-)